scenario_player/tasks/blockchain.py

```diff
@@ -122,16 +133,10 @@ def __init__(
     ) -> None:
         super().__init__(runner, config, parent)
 
-        required_keys = ["contract_name", "event_name", "num_events"]
-        all_required_options_provided = all(key in config.keys() for key in required_keys)
-        if not all_required_options_provided:
-            raise ScenarioError(
-                "Not all required keys provided. Required: " + ", ".join(required_keys)
-            )
-
+        verify_config(config, required_keys=["contract_name", "event_name"])
         self.contract_name = config["contract_name"]
         self.event_name = config["event_name"]
-        self.num_events = config["num_events"]
+        self.num_events = config.get("num_events")
```

Comment:
One thing I just noticed - if this is used in a scenario without the `num_events`, this will pass if there are no events, and also if there are events.
For internal use this makes sense, if one only wants to retrieve the events and not assert anything.